segmentation_models_pytorch/encoders/densenet.py

```diff
@@ -59,37 +45,44 @@ def make_dilated(self, *args, **kwargs):
             "due to pooling operation for downsampling!"
         )
 
-    def get_stages(self):
-        return [
-            nn.Identity(),
-            nn.Sequential(
-                self.features.conv0, self.features.norm0, self.features.relu0
-            ),
-            nn.Sequential(
-                self.features.pool0,
-                self.features.denseblock1,
-                TransitionWithSkip(self.features.transition1),
-            ),
-            nn.Sequential(
-                self.features.denseblock2, TransitionWithSkip(self.features.transition2)
-            ),
-            nn.Sequential(
-                self.features.denseblock3, TransitionWithSkip(self.features.transition3)
-            ),
-            nn.Sequential(self.features.denseblock4, self.features.norm5),
-        ]
-
     def forward(self, x):
-        stages = self.get_stages()
-
-        features = []
-        for i in range(self._depth + 1):
-            x = stages[i](x)
-            if isinstance(x, (list, tuple)):
-                x, skip = x
-                features.append(skip)
-            else:
-                features.append(x)
+        features = [x]
+
+        if self._depth >= 1:
+            x = self.features.conv0(x)
+            x = self.features.norm0(x)
+            x = self.features.relu0(x)
+            features.append(x)
+
+        if self._depth >= 2:
+            x = self.features.pool0(x)
+            x = self.features.denseblock1(x)
+            x = self.features.transition1.norm(x)
+            x = self.features.transition1.relu(x)
+            features.append(x)
+
+        if self._depth >= 3:
+            x = self.features.transition1.conv(x)
+            x = self.features.transition1.pool(x)
+            x = self.features.denseblock2(x)
+            x = self.features.transition2.norm(x)
+            x = self.features.transition2.relu(x)
+            features.append(x)
+
+        if self._depth >= 4:
+            x = self.features.transition2.conv(x)
+            x = self.features.transition2.pool(x)
+            x = self.features.denseblock3(x)
+            x = self.features.transition3.norm(x)
+            x = self.features.transition3.relu(x)
+            features.append(x)
+
+        if self._depth >= 5:
+            x = self.features.transition3.conv(x)
+            x = self.features.transition3.pool(x)
+            x = self.features.denseblock4(x)
+            x = self.features.norm5(x)
+            features.append(x)
```

Comment:
I honestly prefer the for-loop here, but I'm guessing that makes it not possible to compile. What happens if depth > 5? Is there an assert statement to prevent that invalid input?

Reply:
Add validation in https://github.com/qubvel-org/segmentation_models.pytorch/pull/1031/commits/d121fecab27dfa8a54510201e1db8cdfad3d38c1

Reply:
Yes, more like torhcscript limitations, I left lopp in a few places, but it's not possible to loop over indexes, only over layers + `break` is not supported in torchscript